extensions/interactions/base.py

```diff
@@ -89,6 +89,11 @@ class BaseInteraction(object):
     # containing this interaction. These should correspond to names of files in
     # feconf.DEPENDENCIES_TEMPLATES_DIR. Overridden in subclasses.
     _dependency_ids = []
+    # Additional JS library dependencies that should be loaded in pages
```

Comment:
Also we should only be including JS files of prediction service for those interaction which have classifier-data available. If we use dependency_ids then we won't be able to make distinction between the 2 kind of dependencies. We won't be able to know which dependency should be included and which not. So I am not sure if we should combine them.

One thing we can do is make dependency_ids a dict rather than a list and then store 2 kind of dependency in a separate key-value pairs.